silkworm/rpc/core/evm_trace.cpp

```diff
@@ -888,9 +892,38 @@ void TraceTracer::on_execution_end(const evmc_result& result, const silkworm::In
                << " gas_left: " << std::dec << result.gas_left;
 }
 
+void TraceTracer::on_creation_check_failed(const evmc_result& result, const evmc_message& msg) noexcept {
```

Comment:
After we change the callback name to `on_pre_check_failed`, here you can determine anyway the correct value for `trace.type` just using `msg.kind`.

Reply:
Done